downloader.go

```diff
@@ -240,41 +239,53 @@ func (d *Downloader) prepareAndStartDownload(ctx context.Context, url string, wg
 		ch <- s
 		return
 	}
-	defer f.Close()
 	if err := f.Truncate(int64(t)); err != nil {
 		s.Error = fmt.Errorf("error truncating %s to %d: %w", path, t, err)
 		ch <- s
 		return
 	}
+	var urlDownload sync.WaitGroup
+	defer func() {
+		urlDownload.Wait()
+		f.Close()
+	}()
 	for _, c := range d.chunks(t) {
-		b, err := d.downloadChunk(ctx, url, c)
-		if err != nil {
-			s.Error = err
-			ch <- s
-			return
-		}
-		_, err = f.WriteAt(b, int64(c.start))
-		if err != nil {
-			s.Error = fmt.Errorf("error writing to %s: %w", path, err)
+		urlDownload.Add(1)
+		go func(c chunk) {
+			defer urlDownload.Done()
+			b, err := d.downloadChunk(ctx, url, c)
+			if err != nil {
+				s.Error = err
+				ch <- s
+				return
+			}
+			_, err = f.WriteAt(b, int64(c.start))
+			if err != nil {
+				s.Error = fmt.Errorf("error writing to %s: %w", path, err)
+				ch <- s
+				return
+			}
+			s.DownloadedFileBytes += int64(len(b))
```

Comment:
[`c.start` is already `int64`](https://github.com/cuducos/chunk/blob/78be3922b59195aff90ddd5350e545cef6092db7/downloader.go#L90) and maybe we can use the return value of `f.WriteAt` to update `s.DownloadedFileBytes`:

```suggestion
			s, err = f.WriteAt(b, c.start)
			if err != nil {
				s.Error = fmt.Errorf("error writing to %s: %w", path, err)
				ch <- s
				return
			}
			s.DownloadedFileBytes += s
```

Reply:
Done!